src/main/java/com/google/api/codegen/transformer/csharp/CSharpGapicSurfaceTransformer.java

```diff
@@ -0,0 +1,496 @@
+  private final BundlingTransformer bundlingTransformer;
+
+  public CSharpGapicSurfaceTransformer(GapicCodePathMapper pathMapper) {
+    this.pathMapper = pathMapper;
+    this.serviceTransformer = new ServiceTransformer();
+    this.pathTemplateTransformer = new PathTemplateTransformer();
+    this.apiCallableTransformer = new ApiCallableTransformer();
+    this.apiMethodTransformer = new ApiMethodTransformer();
+    this.pageStreamingTransformer = new PageStreamingTransformer();
+    this.bundlingTransformer = new BundlingTransformer();
+    this.importTypeTransformer = new ImportTypeTransformer();
+  }
+
+  @Override
+  public List<ViewModel> transform(Model model, ApiConfig apiConfig) {
+    List<ViewModel> surfaceDocs = new ArrayList<>();
+    SurfaceNamer namer = new CSharpSurfaceNamer(apiConfig.getPackageName());
+
+    for (Interface service : new InterfaceView().getElementIterable(model)) {
+      SurfaceTransformerContext context =
+          SurfaceTransformerContext.create(
+              service, apiConfig, createTypeTable(apiConfig.getPackageName()), namer);
+      addCommonImports(context);
+      StaticLangXApiView xapi = generateXApi(context);
+      StaticLangXSettingsView xsettings = generateXSettings(context);
+
+      surfaceDocs.add(StaticLangXCommonView.newBuilder().api(xapi).settings(xsettings).build());
+    }
+
+    return surfaceDocs;
+  }
+
+  @Override
+  public List<String> getTemplateFileNames() {
+    return Arrays.asList(XAPI_TEMPLATE_FILENAME);
+  }
+
+  private ModelTypeTable createTypeTable(String implicitPackageName) {
+    return new ModelTypeTable(
+        new CSharpTypeTable(implicitPackageName),
+        new CSharpModelTypeNameConverter(implicitPackageName));
+  }
+
+  private StaticLangXApiView generateXApi(SurfaceTransformerContext context) {
+    SurfaceNamer namer = context.getNamer();
+    StaticLangXApiView.Builder xapiClass = StaticLangXApiView.newBuilder();
+    List<StaticLangApiMethodView> methods = generateApiMethods(context);
+
+    xapiClass.doc(serviceTransformer.generateServiceDoc(context, null));
+
+    xapiClass.templateFileName(XAPI_TEMPLATE_FILENAME);
+    xapiClass.packageName(context.getApiConfig().getPackageName());
+    String name = namer.getApiWrapperClassName(context.getInterface());
+    //String grpcName = namer.className(Name.upperCamel(context.getInterface().getSimpleName()));
+    xapiClass.name(name);
+    xapiClass.implName(namer.getApiWrapperClassImplName(context.getInterface()));
+    xapiClass.grpcServiceName(namer.getGrpcContainerTypeName(context.getInterface())); //(grpcName);
+    xapiClass.grpcTypeName(
+        namer.getGrpcServiceClassName(context.getInterface())); //(grpcName + "." + name);
+    xapiClass.settingsClassName(context.getNamer().getApiSettingsClassName(context.getInterface()));
+    List<ApiCallableView> callables =
+        FluentIterable.from(apiCallableTransformer.generateStaticLangApiCallables(context, true))
```

Comment:
I can't tell what the "true" means from this context.

In a separate PR, I'm having @michaelbausor implement a FeatureConfig class which could control things like this. (In his case, he is turning on/off the resource name usage feature.) Could you co-implement FeatureConfig? (in the sense that you create it at the same location, and then we get the two implementations spliced together later). Put it under codegen/transformer/FeatureConfig.java. It should work like SurfaceNamer, where there is a default value for each feature (set to "false"), and then each language turns the features on as they get implemented. For your case:

public boolean mixinsAreOn() {
  return false;
}

You would have a CSharpFeatureConfig; the FeatureConfig instance would be passed to SurfaceTransformerContext when it is created, and then accessed through the context in the code that needs to know.